apps/back/src/Repository/UserRepository.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\User;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\Persistence\ManagerRegistry;
+
+class UserRepository extends ServiceEntityRepository
+{
+
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, User::class);
+    }
+
+    /**
+     * @throws NonUniqueResultException
+     */
+    public function findByEmail(string $email)
+    {
+        return $this->createQueryBuilder('u')
+            ->andWhere('u.email= :email')
+            ->setParameter('email', $email)
+            ->getQuery()->getOneOrNullResult();
```

Comment:
Tu peux directement utiliser `$this->findOneBy(['email' => $email]);` dans ton corps de méthode.